### ISSUE TITLE
CRNN识别训练的问题

### ISSUE THREAD

**Issue by hhhhhhogan** [PaddlePaddle/PaddleOCR#3788]
我使用了150w张图片训练crnn的识别部分，图片都是自己合成、生成的，文字的内容根据新闻、地理知识等语料库的内容进行生成，单行图片的字符数最长为40，图片高度20个像素左右，宽度不定长，crnn_w也设置为320，字典类别为5400左右，目前训练了10个epoch左右，但是结果看上去震荡了，准确率在0.4-0.8之间，请问大概是什么原因造成的呢？如何解决呢？

**Comment by matthew77777** [comment]
Hello 

I think I'm having the simmlar issue.
Could you help me with any advice?

My issue is here 
https://github.com/PaddlePaddle/PaddleOCR/issues/3773

It would be great if you can help me.

Could you share the YAML ?
Thank you so much.

br,

**Comment by hhhhhhogan** [comment]
> Hello
> 
> I think I'm having the simmlar issue.
> Could you help me with any advice?
> 
> My issue is here
> #3773
> 
> It would be great if you can help me.
> 
> Could you share the YAML ?
> Thank you so much.
> 
> br,

hello，I think the amount of data you have is too small, I used 100W of data, but I think it is still too small. Then you can check the classes of the dictionary number and make sure that each word appears in 200-500 imgs(the author recommended). You can check the length of the content of each image, you need to make sure your width of image  enough to be long in order to meet the maxinum lenght, I used the width is 320.Then,I use the default yaml file and do not change it too much.

**Comment by matthew77777** [comment]
Hello 

Thank you so much for the advice.
I used 4500 images but that ammount is not enough ?

Also, "W" of 100 W images is how much ?
Sorry to bother you but please tell me.
Is that 1,000,000 ?

**Comment by hhhhhhogan** [comment]
> Hello
> 
> Thank you so much for the advice.
> I used 4500 images but that ammount is not enough ?
> 
> Also, "W" of "100 W images" is how much ?
> Sorry to bother you but please tell me.
> Is that 1,000,000 ?

yes,100 W images is meaning 1,000,000 images.4500 images are not enough I think.You can check the FAQ in PaddleOCR

**Comment by matthew77777** [comment]
hi 

thank you so much for the advice.
I will get more data and start training again.

Thank you so much.

**Comment by LDOUBLEV** [comment]
> 我使用了150w张图片训练crnn的识别部分，图片都是自己合成、生成的，文字的内容根据新闻、地理知识等语料库的内容进行生成，单行图片的字符数最长为40，图片高度20个像素左右，宽度不定长，crnn_w也设置为320，字典类别为5400左右，目前训练了10个epoch左右，但是结果看上去震荡了，准确率在0.4-0.8之间，请问大概是什么原因造成的呢？如何解决呢？

10 epoch比较短了，训练到60epoch以上看看，
另外你可以加载我们的预训练模型训练

**Comment by hhhhhhogan** [comment]
好的多谢，另外不知道数据量方面会不会少呢？
















在 2021-08-25 10:43:13，"Double_V" ***@***.***> 写道：

我使用了150w张图片训练crnn的识别部分，图片都是自己合成、生成的，文字的内容根据新闻、地理知识等语料库的内容进行生成，单行图片的字符数最长为40，图片高度20个像素左右，宽度不定长，crnn_w也设置为320，字典类别为5400左右，目前训练了10个epoch左右，但是结果看上去震荡了，准确率在0.4-0.8之间，请问大概是什么原因造成的呢？如何解决呢？

10 epoch比较短了，训练到60epoch以上看看，
另外你可以加载我们的预训练模型训练

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by LDOUBLEV** [comment]
> 好的多谢，另外不知道数据量方面会不会少呢？

你的数据量已经足够训练了
